tb_pulumi/secrets.py

```diff
@@ -90,34 +89,28 @@ def __init__(
         opts: pulumi.ResourceOptions = None,
         **kwargs,
     ):
-        super().__init__('tb:secrets:PulumiSecretsManager', name, project, opts=opts, **kwargs)
+        super().__init__('tb:secrets:PulumiSecretsManager', name, project, opts=opts)
         secrets = []
-        versions = []
 
         # First build the secrets
         for secret_name in secret_names:
             # Pull the secret's value from Pulumi's encrypted state
             secret_string = self.project.pulumi_config.require_secret(secret_name)
 
-            # Declare a Secrets Manager Secret
+            # Use our module to build a secret
             secret_fullname = f'{self.project.project}/{self.project.stack}/{secret_name}'
-            secret = aws.secretsmanager.Secret(
-                f'{name}-secret-{secret_name}', opts=pulumi.ResourceOptions(parent=self), name=secret_fullname
+            secret = SecretsManagerSecret(
+                name=f'{name}-{secret_name}',
+                project=self.project,
+                secret_name=secret_fullname,
+                secret_value=secret_string,
+                opts=pulumi.ResourceOptions(parent=self),
+                **kwargs,
             )
             secrets.append(secret)
 
-            # Populate its value
-            versions.append(
-                aws.secretsmanager.SecretVersion(
-                    f'{name}-secretversion-{secret_name}',
-                    opts=pulumi.ResourceOptions(parent=self),
-                    secret_id=secret.id,
-                    secret_string=secret_string,
-                )
-            )
-
         # Then create an IAM policy allowing access to them
-        secret_arns = [secret.arn for secret in secrets]
+        secret_arns = [secret.resources['secret'].arn for secret in secrets]
```

Comment:
Because that `.arn` property is a Pulumi `Output` which works a lot like an async promise, this code won't actually produce a list with values until the ARN value is populated. So either you have valid ARNs and can proceed, or some other step (the step where the Secret or SecretVersion are created, most likely) has failed for some other reason. I don't think we want to try and manually detect errors like this. I think it will wind up obscuring unexpected errors from a more knowledgable library behind a broader one from our comparatively dumb code.